### ISSUE TITLE
Requirements for v10 release

### ISSUE THREAD

**Issue by gfellerph** [swisspost/design-system#4255]
Mind dump of all the things that need to be finished before v10 can go to prod.

- [ ] Complete the [Design System Next Level Milestone](https://github.com/swisspost/design-system/milestone/40). The components in this milestone are required for a production release. Component status will be marked and not every component has to be finished
- [ ] Rewire release action to main again. Due to automations around the release process, creating netlify sites and the decision to develop v9 on the main branch instead of another, we had to rewire prod publications of the docs to the release/v8 branch to be able to still release fixes. This process needs to be reverted in order to release prod docs from main again.
- [ ] Complete the [Remove Bootstrap Milestone](https://github.com/swisspost/design-system/milestone/45). All references to bootstrap need to be removed and the dependency needs to be uninstalled. We've started it, we've promised it, we're almost there, we need to finish it. We can no longer upgrade bootstrap without major investment in manual testing, the token system is not compatible with the bootstrap way of handling color schemes.
- [X] Align `@swisspost/internet-header` versions with the rest of the Design System. During v8, the Internet Header major version was not aligned with the rest of the Design System because it was not necessary to release a major version for this package, which would have had big consequences for all projects using it. With the redesign, a new major version will be released and we can align it with the rest of the Design System. That will be required for the future because there will be a tight dependency on the components package. Fixes for internet-header v1 can be done on the release/v8 branch.
- [ ] Complete the [Migration Milestone](https://github.com/swisspost/design-system/milestone/33). This point will decide on whether projects are willing to upgrade or trying to delay because it seems too much. With a clear and easy to follow instructions, we can ease a lot of the pain that comes with major upgrades of this magnitude. https://github.com/swisspost/design-system/issues/1215 is a reference for breaking changes, but not all breaking changes are listed there
- [x] Automigration solution. Come up with a new way to do automigrations, even if it's just search & replace. Having a tool in place will lower the barrier for upgrading.
- [ ] ~~Header and footer light/dark scheme support. The header and footer do not need to be fully tokenised for v9 to go live (spacings, font size and others), but color tokens should be there to enable dark mode for header and footer~~
- [ ] https://github.com/changesets/changesets/issues/1688
- [ ] https://github.com/swisspost/design-system/issues/5625
- [x] No dependencies on DEV regarding icon uploading/publishing process (input from @rouvenpost )
- [ ] Design/Figma migration tasks (see "Migration guide for Figma" in https://github.com/swisspost/design-system/discussions/5789 ) (input from @sandra-post )
- [ ] https://github.com/swisspost/design-system/issues/5627
- [ ] Test migrations on a real project to see if the bootstrap removal creates unexpected results
- [ ] Cross browser testing of all components
- [ ] Accessibility testing of all components
- [ ] UX review of storybook

**Comment by sandra-post** [comment]
Input @rouvenpost:

Components in [Storybook Next](https://next.design-system.post.ch/?path=/docs/introduction--docs), which a) are still in old Design and need to be in new design and b) need to be in v10:
- List Interactive (still using old icons)
- Popover --> Design and tokens ready
- Stepper --> Design and tokens ready
- Subnavigation (merge with Tabs?) --> Design WIP 
- Tabs ( merge with Tabs?)
- Tag (Design in Backlog) 


Components in [Storybook Next](https://next.design-system.post.ch/?path=/docs/introduction--docs), which a) do not yet have design v2 and b) need discussion whether to be included in v10 or not:
- Collapsible
- Link area
- Pagination
- Progressbar
- Rating
- Skiplinks
- Spinner
- Timepicker
- Toast

**Comment by alizedebray** [comment]
@gfellerph  In my opinion, this list is already too large to maintain a realistic focus.  
I would suggest narrowing it down to the absolute minimum, which in my view is:  

> **Ensuring projects can fully migrate the functionality of existing applications.**  


This means being _Trustworthy_: all components that exist in v9 should also exist in v10, even if they don’t yet have their final v2 look and feel.  

To achieve this, we must provide **at least**:  
- **Carousel:** [~10 files on Git IT](https://gitit.post.ch/plugins/servlet/search?q=%22ngb-carousel%22%20lang%3Ahtml)  
- **Datepicker:** [~400 files on Git IT](https://gitit.post.ch/plugins/servlet/search?q=%22ngb-datepicker%22%20lang%3Ahtml)  
- **Pagination:** [~230 files on Git IT](https://gitit.post.ch/plugins/servlet/search?q=%22ngb-pagination%22%20lang%3Ahtml)  
- **Progress bar:** [~50 files on Git IT](https://gitit.post.ch/plugins/servlet/search?q=%22ngb-progressbar%22%20lang%3Ahtml)  
- **Horizontal stepper:** [~30 files on Git IT](https://gitit.post.ch/plugins/servlet/search?q=%22stepper-item%22%20lang%3Ahtml)  
- **Subnavigation:** [~45 files on Git IT](https://gitit.post.ch/plugins/servlet/search?q=%22subnavigation-item%22%20lang%3Ahtml)  
- **Timepicker:** [~50 files on Git IT](https://gitit.post.ch/plugins/servlet/search?q=%22ngb-timepicker%22%20lang%3Ahtml)  
- **Typeahead:** [~240 files on Git IT](https://gitit.post.ch/plugins/servlet/search?q=%22ngbTypeahead%22%20lang%3Ahtml)  
- **Header generated from a configuration object**  
- **Migration guidelines** (even if not yet user-friendly)  

If we also want to respect our principle of _“Easy to use”_, we should add:  
- **Understandable change logs**  
- **Basic auto-migration** (we can expand this later)  

Estimating these tasks will allow us to define a minimum release date.  
- Any release date before that would mean v10 is not production ready.  
- Any release date after that allows us to add additional nice-to-have features from the larger list.  

For context: the composable header alone took almost 4 months to complete. I expect the configurable header to take several weeks as well. Adding the other tasks on top, along with support and bug fixing, could realistically bring us to the end of the year.  

If we finish earlier, that’s even better—it simply means we can include more features in the release!

**Comment by sandra-post** [comment]
@gfellerph : @rouvenpost and myself have created an [Excel overview](https://postchag.sharepoint.com.mcas.ms/:x:/r/sites/cwf/_layouts/15/doc.aspx?sourcedoc=%7B2616c268-cb35-4b34-a758-665f7022bc36%7D&action=edit) of the current component (and storybook) status including a proposal if compo should be ready for release or not.

Our list of proposed and yet to be done compos is the same as the one from @alizedebray above, plus:
- File upload
- Inline notification (Process notification)
- Tabs

**Comment by sandra-post** [comment]
@sandra-post Create tickets for Design-related migration tasks in release v10 + excel

**Comment by gfellerph** [comment]
@alizedebray check if all tickets have been created, then close the issue.
